server/src/main/java/org/elasticsearch/action/search/ReduceSearchPhaseException.java

```diff
@@ -27,4 +29,21 @@ public ReduceSearchPhaseException(String phaseName, String msg, Throwable cause,
     public ReduceSearchPhaseException(StreamInput in) throws IOException {
         super(in);
     }
+
+    @Override
+    public RestStatus status() {
+        ShardSearchFailure[] shardFailures = getShardFailures();
+        if (shardFailures.length == 0) {
+            return getCause() == null ? RestStatus.INTERNAL_SERVER_ERROR : ExceptionsHelper.status(getCause());
+        }
+        RestStatus status = shardFailures[0].status();
+        if (shardFailures.length > 1) {
+            for (int i = 1; i < shardFailures.length; i++) {
+                if (shardFailures[i].status().getStatus() >= 500) {
```

Comment:
I am not a big fan of the hardcoded `500` value here. 
Is there any way to add a constant such as `RestStatus.INTERNAL_SERVER_ERROR.getStatus()`

Also, should you perhaps delegate this to the `RestStatus.status()` method?

Would running something like `RestStatus.status(0, shardFailures.length, shardFailures)` return the same result? Again, I am not an expert on this part of the code, so take this with a grain of salt.